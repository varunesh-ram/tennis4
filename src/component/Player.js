import React from 'react';
import PropTypes from 'prop-types';

export default class Player extends React.Component {
    render() {
        return (
            <div>
                <h5>{this.props.name}</h5>
            </div>
        );
    }
}

Player.propTypes = {
    name: PropTypes.string.isRequired
}