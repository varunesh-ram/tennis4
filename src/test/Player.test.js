import React from 'react';
import { shallow } from 'enzyme';
import Player from '../component/Player';

describe(("<Player/> component with props"), () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(< Player name="Player 1" />);
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should have passed heading from props", () => {
    expect(wrapper.find("h5").text()).toEqual("Player 1");
  });
});