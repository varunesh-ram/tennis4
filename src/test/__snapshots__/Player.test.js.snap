// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player/> component with props should render correctly 1`] = `ShallowWrapper {}`;
